spec/services/requests_total_items_service_spec.rb

```diff
@@ -0,0 +1,32 @@
+RSpec.describe RequestsTotalItemsService, type: :service do
+  describe '#calculate' do
+    it 'return items with quantities calculated' do
+      items = Item.active.sample(3)
+      items_id = items.pluck(:id)
+      requests = [
+        create(:request, request_items: request_items(items_id, 20)),
+        create(:request, request_items: request_items(items_id, 10))
+      ]
+
+      items_calculated = described_class.calculate(requests)
+      expect(total_first_item(items_calculated)).to eq(30)
+    end
+
+    it 'return blank when requests itens is blank' do
+      request = create(:request, request_items: {})
+      expect(described_class.calculate([request])).to be_blank
+    end
+
+    it 'return blank when requests is nil' do
+      expect(described_class.calculate(nil)).to be_blank
+    end
```

Comment:
There is an alternative more rspec'y way to do this that conforms a little nicer to the current partner. Here is another way that I would implement it.
```
RSpec.describe RequestsTotalItemsService, type: :service do
  describe '.calculate' do
    subject { described_class.calculate(requests) }

    context 'when the request items is not blank' do
      let(:requests) do
        items = Item.active.sample(3)
        item_ids = items.pluck(:id)

        [
          create(:request, request_items: item_ids.map { |k| { "item_id" => k, "quantity" => 20 } }),
          create(:request, request_items: item_ids.map { |k| { "item_id" => k, "quantity" => 10 } })
        ]
      end

      it 'return items with correct quantities calculated' do
        expect(subject.first.last).to eq(30)
      end
    end

    context 'when provided with requests that have no request items' do
      let(:requests) { [create(:request, request_items: {})] }

      it 'return blank when requests itens is blank' do
        expect(subject).to be_blank
      end
    end

    context 'when provided requests is nil' do
      let(:requests) { nil }

      it 'return blank when requests itens is blank' do
        expect(subject).to be_blank
      end
    end
  end
end
```
The differences that I want to highlight are:
- It uses `let` to define the context and defines a `subject` clearly
- It splits out the context from the assertion (it)
- Removed the methods and kept them to the spot were they were used (in the first context/assertion)

If you don't feel strongly, you can copy & paste the code above. Specs pass when I tested it out.

Let me know if you have any Qs :).